### PR TITLE
APB-1880][RV,MP] Change invitations core flow to include MTD-VAT

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/audit/AuditService.scala [hunk 1]
```diff
@@ -46,8 +46,8 @@ class AuditService @Inject() (val auditConnector: AuditConnector) {
         "invitationId" -> invitationId,
         "agentReferenceNumber" -> arn.value,
         "clientIdType" -> "ni",
-        "clientId" -> agentInvitationUserInput.nino.value,
-        "service" -> agentInvitationUserInput.service.getOrElse(throw new IllegalStateException("No regime present"))
```
Comment (JTY009) [inline app/uk/gov/hmrc/agentinvitationsfrontend/audit/AuditService.scala]: Why is this removed?

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala [hunk 2]
```diff
@@ -77,17 +80,22 @@ class AgentsInvitationController @Inject()(
...
     }
   }
 
   val selectService: Action[AnyContent] = Action.async { implicit request =>
     withAuthorisedAsAgent { arn =>
-      request.session.get("nino") match {
-        case Some(nino) => Future successful Ok(select_service(agentInvitationServiceForm.fill(AgentInvitationUserInput(Nino(nino), None, None)),
-          personalIncomeRecord ++ mtdItId))
-        case _ => Future successful Redirect(routes.AgentsInvitationController.showNinoForm())
+      request.session.get("service") match {
```
Comment (JTY009) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala]: Could line this be removed because the user has not selected an option yet?

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationControllerISpec.scala [hunk 3]
```diff
@@ -123,49 +174,38 @@ class AgentInvitationControllerISpec extends BaseISpec {
...
   "POST /agents/select-service" should {
     val request = FakeRequest("POST", "/agents/select-service")
     val submitService = controller.submitService()
 
-    "return 303 for authorised Agent with valid nino and ITSA service, redirect to postcode page" in {
-      val serviceForm = agentInvitationServiceForm.fill(AgentInvitationUserInput(validNino, Some(serviceITSA), None))
-      val result = submitService(authorisedAsValidAgent(request.withFormUrlEncodedBody(serviceForm.data.toSeq: _*)
-        .withSession("nino" -> validNino.value), arn.value))
+    "return 303 for authorised Agent with valid ITSA service, redirect to enter nino page" in {
+      val serviceForm = agentInvitationServiceForm.fill(AgentInvitationUserInput(serviceITSA, Some(validNino), None))
```
Comment (JTY009) [inline it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationControllerISpec.scala]: validNino is not necessary in this test i think?

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationControllerISpec.scala [hunk 4]
```diff
@@ -123,49 +174,38 @@ class AgentInvitationControllerISpec extends BaseISpec {
...
 
-    "return 303 for authorised Agent with valid nino and Personal Income Record service, redirect to invitation sent page" in {
-      createInvitationStubForPIR(arn, validNino.value, invitationId, validNino.value, servicePIR, "NI")
-      getInvitationStub(arn, validNino.value, invitationId, servicePIR, "NI","Pending")
-
-      val serviceForm = agentInvitationServiceForm.fill(AgentInvitationUserInput(validNino, Some(servicePIR), None))
-      val result = submitService(authorisedAsValidAgent(request.withFormUrlEncodedBody(serviceForm.data.toSeq: _*)
-        .withSession("nino" -> validNino.value), arn.value))
+    "return 303 for authorised Agent with valid Personal Income Record service, redirect to enter nino" in {
+      val serviceForm = agentInvitationServiceForm.fill(AgentInvitationUserInput(servicePIR, Some(validNino), None))
```
Comment (JTY009) [inline it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationControllerISpec.scala]: same here

File: app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala [hunk 5]
```diff
@@ -31,8 +31,8 @@ class InvitationsService @Inject() (invitationsConnector: InvitationsConnector,
                                     agentServicesAccountConnector: AgentServicesAccountConnector) {
 
   def createInvitation(arn: Arn, userInput: AgentInvitationUserInput)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Invitation] = {
-    val service = userInput.service.getOrElse(throw new Exception("Service is missing"))
```
Comment (JTY009) [inline app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala]: Why is it removed?

File: app/uk/gov/hmrc/agentinvitationsfrontend/audit/AuditService.scala (hunk 1 above)
Comment (rvijayan) [reply app/uk/gov/hmrc/agentinvitationsfrontend/audit/AuditService.scala]: We haven't removed it. Necessary changes has been made to include  nino option.

File: app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala (hunk 5 above)
Comment (rvijayan) [reply app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala]: Its not removed. The `service` is not optional now.

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala (hunk 2 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala]: Ok

File: app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala (hunk 5 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentinvitationsfrontend/services/InvitationsService.scala]: Never mind

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala (hunk 2 above)
Comment (rvijayan) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala]: Ignore me, you are right. Fixing the issue now..
